mm/routines/product.py

```diff
@@ -17,11 +17,18 @@ def update_enum_product_attribute(self, sku, att_name, att_value):
         return record
 
     def get_configurable_associated_products(self, store_id=None, is_active=None, is_visible=None):
+        legal_inputs = set([None, True, False])
+        message = "Not a valid input value for '{0}'. Use: {1}"
+
+        assert type(store_id) is int or store_id is None, message.format('store_id', 'None or int')
+        assert is_active in legal_inputs, message.format('is_active', 'None, True or False')
```

Comment:
You're missing it.

See (the primary matter of these are fairly short):

http://stackoverflow.com/a/134659/706421
http://stackoverflow.com/a/1085656/706421
http://stackoverflow.com/a/133024/706421


While you're reading these, remember that there are many other different
implementations of Python than cPython (ours): e.g. PyPy, Cython,
IronPython, Jython


On Thu, May 19, 2016 at 10:11 AM, Diego Terzano <notifications@github.com>
wrote:

> In mm/routines/product.py
> <https://github.com/CoffeeForThinkers/MagentoModels/pull/4#discussion_r63885062>
> :
>
> > @@ -17,11 +17,18 @@ def update_enum_product_attribute(self, sku, att_name, att_value):
> >          return record
> >
> >      def get_configurable_associated_products(self, store_id=None, is_active=None, is_visible=None):
> > +        legal_inputs = set([None, True, False])
> > +        message = "Not a valid input value for '{0}'. Use: {1}"
> > +
> > +        assert type(store_id) is int or store_id is None, message.format('store_id', 'None or int')
> > +        assert is_active in legal_inputs, message.format('is_active', 'None, True or False')
>
> I get that, In Python boolis a subclass of an int. There should be no
> difference between 1 and True or 0 and False.
> I could have done is_active in [1,0,None]
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/CoffeeForThinkers/MagentoModels/pull/4/files/9aa86cb203d8cf0a259523bfe5481f93b0514543..b90e3dbb8e234665fe945101d8b32cb84d373d25#r63885062>
>